torch/testing/_internal/common_nn.py

```diff
@@ -4934,7 +4950,7 @@ def test_cuda(self, test_case):
             raise unittest.SkipTest('Excluded from CUDA tests')
 
         cpu_input = self._get_input()
-        type_map = {'torch.DoubleTensor': torch.cuda.FloatTensor}  # type: ignore[attr-defined]
+        type_map = {'torch.DoubleTensor': torch.FloatTensor}  # type: ignore[attr-defined]
```

Comment:
Why is this not on cuda anymore?

Reply:
we still call `to_gpu` afterwards, so whether this is `torch.FloatTensor` or `torch.cuda.FloatTensor` shouldn't matter. The key is that doubles are mapped to floats, and I thought removing `cuda` here will make that more clear

Reply:
> we still call to_gpu afterwards

This map is actually used in that call right? And you added in this PR the `.cuda()` in that function that forces the move to gpu.

I think this is ok to do, maybe add a comment in the `to_gpu` function to mention that the type map should be only between cpu types?
Or maybe even better, change the type_map to contain `dtypes` instead of string/class. That way you cannot do device movement by mistake :D

Reply:
Yep, changing type_map to be between dtypes

Reply:
This also allows me to do `to(dtype=t, device="cuda")`